packages/studiocms_dashboard/src/components/Layout.astro

```diff
@@ -1,32 +1,37 @@
 ---
-import { ViewTransitions } from 'astro:transitions';
+import { ClientRouter } from 'astro:transitions';
 import { Button } from '@studiocms/ui/components';
 import Icon from '@studiocms/ui/utils/Icon.astro';
 import { BaseHead } from '../components';
 import SingleSidebar from './SingleSidebar.astro';
+import LoginChecker from './islands/LoginChecker.astro';
 
 type Props = {
 	title: string;
 	description: string;
-	includeSidebar?: boolean;
-	sideBarActiveItemID: string;
+	sidebar?: boolean | 'single' | 'double';
+	requiredPermission?: 'owner' | 'admin' | 'editor' | 'visitor' | 'unknown' | 'none';
 };
 
-const { title, description } = Astro.props;
+const { title, description, requiredPermission = 'unknown', sidebar = 'single' } = Astro.props;
 ---
 
 <!DOCTYPE html>
-<html lang="en" data-theme={'dark'}>
+<html lang="en">
   <head>
     <BaseHead title={title} description={description} />
-    <ViewTransitions />
+    <ClientRouter />
   </head>
   <body>
-    <SingleSidebar transition:persist transition:persist-props />
+    { requiredPermission !== 'none' && <LoginChecker server:defer {requiredPermission} /> }
+    { sidebar === 'single' && <SingleSidebar transition:persist transition:persist-props /> }
+    <!-- { sidebar === 'double' && <DoubleSidebar transition:persist transition:persist-props /> } -->
```

Comment:
Why is the double sidebar commented out?

Reply:
... this PR is to another unfinished pr ( #393 ) :sweat_smile: just moved to another branch to make sure i didn't screw the pooch during the v5 upgrade and had a good restore point... but it went way better than i had thought it would